src/daemon/bitcoind/actions.rs

```diff
@@ -323,6 +323,44 @@ fn mark_confirmed_spends(
     Ok(())
 }
 
+fn mark_confirmed_cancels(
+    revaultd: &Arc<RwLock<RevaultD>>,
+    bitcoind: &BitcoinD,
+) -> Result<(), BitcoindError> {
```

Comment:
As mentioned for the DB schema i think  here you just want to iterate over the Cancel transaction (or maybe a tuple of Vault, Cancel) and ask bitcoind about the onchain state of each Cancel txid.